request-manager/graph/grapher.go

```diff
@@ -272,13 +294,14 @@ func buildSeqGraph(seqSpec *spec.Sequence, idgen id.Generator) (seqGraph *Graph,
 
 				delete(nodesToAdd, nodeName)
 				nodesAdded[nodeName] = true
-				componentAdded = true
+				nodeAdded = true
```

Comment:
I don't think this is needed, you can use `length(nodesAdded)` instead?

Reply:
The code used to save `len(nodesAdded)` for this purpose instead of using a flag. I realize that it was more efficient that way because we don't need to update `nodesAdded` on every iteration of this inner loop over nodesToAdd, but it was also somewhat less readable to me, since that was another var we needed to track from outside the outer loop.

Reply:
We can resolve this. I think the other comment about detecting a cycle in A->B->C addressed this.